modules/stochastic_tools/src/samplers/SobolSampler.C

```diff
@@ -0,0 +1,57 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "SobolSampler.h"
+
+template <>
+InputParameters
+validParams<SobolSampler>()
+{
+  InputParameters params = validParams<Sampler>();
+  params.addClassDescription("Sobol variance-based sensitivity analysis Sampler.");
+  params.addRequiredParam<unsigned int>(
+      "n_samples", "Number of Monte Carlo samples to perform for each distribution.");
+  return params;
+}
+
+SobolSampler::SobolSampler(const InputParameters & parameters)
+  : Sampler(parameters),
+    _num_samples(getParam<unsigned int>("n_samples")),
+    _a_matrix(0, 0),
+    _b_matrix(0, 0)
+{
+  setNumberOfRequiedRandomSeeds(2);
+}
+
+void
+SobolSampler::sampleSetUp()
+{
+  _a_matrix.resize(_num_samples, _distributions.size());
+  _b_matrix.resize(_num_samples, _distributions.size());
+  for (unsigned int i = 0; i < _num_samples; ++i)
+    for (unsigned int j = 0; j < _distributions.size(); ++j)
```

Comment:
In some places you are using `std::size_t` and others `unsigned int` - how about `auto`? 😉